synapse/handlers/oidc.py

```diff
@@ -935,17 +952,32 @@ async def handle_redirect_request(
 
```

Comment:
n.b. The docstring is now incomplete because it fails to mention the code challenge being included in the redirect URI and the code verifier being included in the macaroon (or even PKCE support at all).

Reply:
I'll clarify this. Thanks!

Reply:
Hopefully the changes in 1e82afd7e88d9194b368519ecd696adbca4c856f make sense.